src/main/java/org/snomed/snowstorm/core/data/services/ConceptUpdateHelper.java

```diff
@@ -598,28 +598,27 @@ private void doSaveBatchQueryConcept(Collection<QueryConcept> queryConcepts, Com
 			if (existingComponent != null) {
 				newComponent.setCreating(false);// May have been set true earlier
 				newComponent.copyReleaseDetails(existingComponent, rebaseParentExistingComponentMap.get(newComponent.getId()));
-				newComponent.updateEffectiveTime();
 			} else {
 				newComponent.setCreating(true);
 				newComponent.clearReleaseDetails();
 			}
-			
-			
-			//Any change to a component in an extension needs to be done in the default module
-			if (newComponent.isChanged() && (defaultModuleId != null && 
-					!defaultModuleId.equals(Concepts.CORE_MODULE))) {
+
+			// Any change to a component in an extension needs to be done in the default module
+			if (newComponent.isComponentChanged(existingComponent) && (defaultModuleId != null && !defaultModuleId.equals(Concepts.CORE_MODULE))) {
 				newComponent.setModuleId(defaultModuleId);
 			}
+			newComponent.updateEffectiveTime();
 
 			// Trying Concept module in attempt to restore effective time for the case
 			// where content has changed and then been reverted.
-			if (shouldRestoreDefaultModuleId(defaultModuleId, existingComponent, newComponent)) {
```

Comment:
@astro-snail  This fix seems to have fixed your bug but there are other cases which are still not correct. e.g new components created on international are having null module id. I have listed test scenarios in a google sheet and will share with you.